### ISSUE TITLE
Import named 'diffieHellman' not found in module 'crypto'

### ISSUE THREAD

**Issue by huseeiin** [oven-sh/bun#6782]
### What version of Bun is running?

1.0.7

### What platform is your computer?

Linux 6.1.0-10-amd64 x86_64 unknown

### What steps can reproduce the bug?

use jose in vite as an ssr module

### What is the expected behavior?

it works in node

### What do you see instead?

SyntaxError: Import named 'diffieHellman' not found in module 'crypto'.
    at processTicksAndRejections (:61:39)

### Additional information

```ts
import {EncryptJWT, jwtDecrypt} from 'jose'

const SECRET = new TextEncoder().encode('SECRET')

const encrypted = await new EncryptJWT({hello: 'world'})
	.setProtectedHeader({
		alg: 'dir',
		enc: 'A128CBC-HS256'
	})
	.setExpirationTime('1h')
	.encrypt(SECRET)

const decrypted = await jwtDecrypt(encrypted, SECRET)
```
some code to test

**Comment by SaimonWoidig** [comment]
Using jose in Astro SSR (using Vite SSR modules). Bun version is 1.0.7 aswell.

```
bun -version
1.0.7

bunx --bun astro dev --host
  🚀  astro  v3.4.0 started in 249ms
  
  ┃ Local    http://localhost:4321/
  ┃ Network  http://172.20.128.76:4321/
  
2:36:33 PM [vite] Error when evaluating SSR module /home/swoidig/programming/my-astro-project-using-bun/src/middleware: failed to import "jose"
|- SyntaxError: Import named 'diffieHellman' not found in module 'crypto'.
    at processTicksAndRejections (:61:39)
```

Also I am only using the `jwtVerify` method. I think that simply using the module in a Vite SSR module running on Bun causes the error.

**Comment by huseeiin** [comment]
> Using jose in Astro SSR (using Vite SSR modules). Bun version is 1.0.7 aswell.
> 
> ```
> bun -version
> 1.0.7
> 
> bunx --bun astro dev --host
>   🚀  astro  v3.4.0 started in 249ms
>   
>   ┃ Local    http://localhost:4321/
>   ┃ Network  http://172.20.128.76:4321/
>   
> 2:36:33 PM [vite] Error when evaluating SSR module /home/swoidig/programming/my-astro-project-using-bun/src/middleware: failed to import "jose"
> |- SyntaxError: Import named 'diffieHellman' not found in module 'crypto'.
>     at processTicksAndRejections (:61:39)
> ```
> 
> Also I am only using the `jwtVerify` method. I think that simply using the module in a Vite SSR module running on Bun causes the error.

nice to see i'm not the only one. i was using sveltekit btw :D

**Comment by cirospaciari** [comment]
@xxxhussein this is still happening on bun v1.0.11? 

now running the test code I got this result:
```js

import { EncryptJWT, jwtDecrypt } from "jose";

const SECRET = new TextEncoder().encode("12345678912345678912345678912345");

const encrypted = await new EncryptJWT({ hello: "world" })
  .setProtectedHeader({
    alg: "dir",
    enc: "A128CBC-HS256",
  })
  .setExpirationTime("1h")
  .encrypt(SECRET);

const decrypted = await jwtDecrypt(encrypted, SECRET);
console.log(decrypted);
```
```js
{
  payload: {
    hello: "world",
    exp: 1699567220
  },
  protectedHeader: {
    alg: "dir",
    enc: "A128CBC-HS256"
  }
}
```

**Comment by SaimonWoidig** [comment]
> @xxxhussein this is still happening on bun v1.0.11?
> 
> now running the test code I got this result:
> 
> ```js
> import { EncryptJWT, jwtDecrypt } from "jose";
> 
> const SECRET = new TextEncoder().encode("12345678912345678912345678912345");
> 
> const encrypted = await new EncryptJWT({ hello: "world" })
>   .setProtectedHeader({
>     alg: "dir",
>     enc: "A128CBC-HS256",
>   })
>   .setExpirationTime("1h")
>   .encrypt(SECRET);
> 
> const decrypted = await jwtDecrypt(encrypted, SECRET);
> console.log(decrypted);
> ```
> 
> ```js
> {
>   payload: {
>     hello: "world",
>     exp: 1699567220
>   },
>   protectedHeader: {
>     alg: "dir",
>     enc: "A128CBC-HS256"
>   }
> }
> ```

I have to try it on my own. But since you tried it, I guess it works now. I will try it out as soon as I can.

**Comment by huseeiin** [comment]
> @xxxhussein this is still happening on bun v1.0.11?
> 
> now running the test code I got this result:
> 
> ```js
> import { EncryptJWT, jwtDecrypt } from "jose";
> 
> const SECRET = new TextEncoder().encode("12345678912345678912345678912345");
> 
> const encrypted = await new EncryptJWT({ hello: "world" })
>   .setProtectedHeader({
>     alg: "dir",
>     enc: "A128CBC-HS256",
>   })
>   .setExpirationTime("1h")
>   .encrypt(SECRET);
> 
> const decrypted = await jwtDecrypt(encrypted, SECRET);
> console.log(decrypted);
> ```
> 
> ```js
> {
>   payload: {
>     hello: "world",
>     exp: 1699567220
>   },
>   protectedHeader: {
>     alg: "dir",
>     enc: "A128CBC-HS256"
>   }
> }
> ```

yes

**Comment by huseeiin** [comment]
it works now. I was on  v1.0.11 and upgraded to v1.0.13 so idk if the fix was in v1.0.12 or v1.0.13

**Comment by ssebastianoo** [comment]
I'm on SvelteKit using Bun 1.1.12 and I still get `SyntaxError: Export named 'diffieHellman' not found in module 'crypto'.`
![image](https://github.com/oven-sh/bun/assets/48064277/81354332-99d9-4834-a8ed-e35d91c37c32)

**Comment by huseeiin** [comment]
> I'm on SvelteKit using Bun 1.1.12 and I still get `SyntaxError: Export named 'diffieHellman' not found in module 'crypto'.` ![image](https://private-user-images.githubusercontent.com/48064277/338214348-81354332-99d9-4834-a8ed-e35d91c37c32.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMzkwOTksIm5iZiI6MTcxODEzODc5OSwicGF0aCI6Ii80ODA2NDI3Ny8zMzgyMTQzNDgtODEzNTQzMzItOTlkOS00ODM0LWE4ZWQtZTM1ZDkxYzM3YzMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDIwNDYzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE5Njg2ODA5Njc4MmQyYTYxMTc3ZDc0OWM2NjI2MDgwYWUyNjY2NTdhMjc4ZWMyZTY1NWQxMjJmMzBlMjkxMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kXjK6EMHH_-4Hcb2y--R5lbJJtMvV3doZcDbp4wxj5Q)

i guess i should reopen

**Comment by birkskyum** [comment]
There is an implementation here in rust - https://crates.io/crates/x25519-dalek don't know if that could be used.

**Comment by PaddeK** [comment]
Same issue here while running aws cdk with bun. 

bun: 1.1.20
cdk: 2.148.0 (build e5740c0)

**Comment by birkskyum** [comment]
That's unfortunate, considering the 1.1.19 announcement of [aws-cdk-lib support](https://bun.sh/blog/bun-v1.1.19)

**Comment by birkskyum** [comment]
Just looking at reported cases, this appear to be troubling people using Next, Nuxt, SvelteKit and SolidStart, but likely more.

**Comment by huseeiin** [comment]
> Just looking at reported cases, this appear to be troubling people using Next, Nuxt, SvelteKit and SolidStart, but it's likely even more are affected.

yes, what all these have in common is they're vite SSR frameworks

**Comment by marvinvr** [comment]
I've experienced this issue when using [AuthJS](https://authjs.dev/reference/sveltekit) together with SvelteKit. It only happened when I built and ran the app using bun though, not during development.

**Comment by sbscan** [comment]
> I've experienced this issue when using [AuthJS](https://authjs.dev/reference/sveltekit) together with SvelteKit. It only happened when I built and ran the app using bun though, not during development.

same here

**Comment by rikusen0335** [comment]
Me, as same issue happen, was not installing nodejs and results `which node` outputs nothing.
If this help someone
